skyportal/tests/api/test_taxonomy.py

```diff
@@ -0,0 +1,85 @@
+
+
+def test_add_retrieve_delete_taxonomy(taxonomy_token, public_group):
+    status, data = api('POST', 'taxonomy',
+                       data={
+                             'name': "test taxonomy",
+                             'hierarchy': taxonomy,
+                             'group_ids': [public_group.id],
+                             'provenance': f"tdtax_{__version__}",
+                             'version': __version__,
+                             'isLatest': True
+                             },
+                       token=taxonomy_token)
+    assert status == 200
+    taxonomy_id = data['data']['taxonomy_id']
+
+    status, data = api('GET', f'taxonomy/{taxonomy_id}', token=taxonomy_token)
+
+    assert status == 200
+    assert data['data']['name'] == 'test taxonomy'
+    assert data['data']['version'] == __version__
+
+    status, data = api('DELETE', f'taxonomy/{taxonomy_id}', token=taxonomy_token)
+    assert status == 200
+
+
+def test_add_bad_taxonomy(taxonomy_token, public_group):
+    status, data = api('POST', 'taxonomy',
+                       data={
+                             'name': "test bad taxonomy",
+                             'hierarchy': {"Silly": "taxonomy", "bad": True},
+                             'group_ids': [public_group.id],
+                             'provenance': "Nope",
+                             'version': "0.0.1bad",
+                             'isLatest': True
+                             },
+                       token=taxonomy_token)
+
+    assert status == 400
+    assert data['message'] == "Hierarchy does not validate against the schema."
+
+
+def test_latest_taxonomy(taxonomy_token, public_group):
+
+    # add one, then add another with the same namer
+    status, data = api('POST', 'taxonomy',
+                       data={
+                             'name': "test taxonomy",
+                             'hierarchy': taxonomy,
+                             'group_ids': [public_group.id],
+                             'provenance': f"tdtax_{__version__}",
+                             'version': __version__
+                             },
+                       token=taxonomy_token)
+    assert status == 200
+    old_taxonomy_id = data['data']['taxonomy_id']
+    status, data = api('GET', f'taxonomy/{old_taxonomy_id}',
+                       token=taxonomy_token)
+    assert status == 200
+    assert data['data']['isLatest']
+
+    status, data = api('POST', 'taxonomy',
+                       data={
+                             'name': "test taxonomy",
+                             'hierarchy': taxonomy,
+                             'group_ids': [public_group.id],
+                             'provenance': f"tdtax_{__version__}",
+                             'version': "new version"
+                             },
+                       token=taxonomy_token)
+    assert status == 200
+    new_taxonomy_id = data['data']['taxonomy_id']
```

Comment:
should we also check that `isLatest == True` here?

Reply:
Done